src/main/java/cz/cvut/kbss/termit/persistence/dao/changetracking/ChangeRecordDao.java

```diff
@@ -58,13 +62,16 @@ public List<AbstractChangeRecord> findAll(Asset<?> asset) {
         Objects.requireNonNull(asset);
         try {
             final Descriptor descriptor = new EntityDescriptor();
+            URI changeTrackingContextUri = changeTrackingContextResolver.resolveChangeTrackingContext(asset);
             descriptor.setLanguage(null);
             return em.createNativeQuery("SELECT ?r WHERE {" +
+                             "GRAPH ?ctc { ?r ?relatesTo ?asset . }" +
                              "?r a ?changeRecord ;" +
                              "?relatesTo ?asset ;" +
```

Comment:
This can (and should) be removed from the query since it is already in the `GRAPH` part. Or, you can wrap the whole query in `GRAPH`. It may perform better.
```java
"SELECT ?r WHERE {" +
"GRAPH ?ctc {" +
"a ?changeRecord ;" +
"?relatesTo ?asset ;" +
"?hasTime ?timestamp ." +
"OPTIONAL { ?r ?hasChangedAttribute ?attribute . } " +
} } ORDER BY DESC(?timestamp) ?attribute"
```

Reply:
I tried wrapping the whole query in named GRAPH, but then it did not resolve that `vytvoření-entity` and `úprava-entity` are subclasses of `změna`, which the query searches for. But most certainly, I will optimize it.